File: support/ruby/env.rb
require 'cucumber/core/test/filters/activate_steps_for_self_test'

Before('@failing-before') do
  raise "Failing before hook"
end

Given /pass/ do
end

Given /fail/ do
  fail "Fail"
end

Given /pending/ do
  pending
end

Given /skip/ do
  skip_this_scenario
end

File: support/js/env.js
var { Given, Before } = require('cucumber')

Given(/pass/, function () {
})

Given(/fail/, function () {
  throw new Error("this step failed")
})

Given(/skip/, function () {
  return 'skipped'
})

Before({ tags: "@failing-before" }, function () {
  throw new Error("failing before hook")
})
